app/controllers/mixins/ems_common_angular.rb

```diff
@@ -713,8 +713,13 @@ def build_credentials(ems, mode)
     end
 
     def retrieve_event_stream_selection
-      return "amqp" if @ems.connection_configurations.ceilometer.try(:endpoint).nil? && @ems.connection_configurations.amqp.try(:endpoint)
-      "ceilometer"
+      if @ems.connection_configurations.ceilometer.try(:endpoint).nil? && @ems.connection_configurations.amqp.try(:endpoint)
+        amqp_auth = @ems.authentications.detect { |a| a.authtype == 'amqp' }
+
+        return 'none' if amqp_auth.nil?
+        return 'amqp'
+      end
+      'ceilometer'
```

Comment:
I don't like that 'ceilometer' is fallback when nothing else is true, can you please reorganize function to following shape:

```
return 'amqp' if ...
return 'ceilometer' if ...
return 'none'
```

Reply:
Also, please add unit test for this function, e.g. in a new file `spec/controllers/mixins/ems_common_angular_spec.rb`.